File: contests/lc/2900.LongestUnequalAdjacentGroupsSubsequenceI/longest_neq_group.go
package _2900_longest_neq_group

func getLongestSubsequence(words []string, groups []int) (ans []string) {
	for i, x := range groups {
		if i == 0 || x != groups[i-1] {
			ans = append(ans, words[i])
		}
	}
	return
}

File: contests/lc/2900.LongestUnequalAdjacentGroupsSubsequenceI/longest_neq_group.py
from typing import List


class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        ans: List[str] = []
        prev_group = None
        for word, group in zip(words, groups):
            if prev_group is None or group != prev_group:
                ans.append(word)
                prev_group = group
        return ans
